### DISCUSSION TITLE
Clarification on `Figure`, `SimpleFigure` & `Table`

### DISCUSSION THREAD

**Discussion by argent0** [tarleb/pandoc#26] [General]
Hi, just to make clear what are the concepts we have introduced so far in this project, I would like to clarify what each of the elements on the title are.

# `Figure`

```{haskell}
data  Block
  ...
  | Figure Attr Caption [Block]
  ...
```

Represents a group of Blocks inside the Document. This could be considered a document within another document. It has attributes and a caption.

Related to LaTeX's `\begin{figure}` and HTML5's `<figure>`

# `SimpleFigure`

```
SimpleFigure attr plainTxt ("src", "title")
```

It doesn't introduce a new Block constructor. It's a synonym for:

```
Para [Image attr plainTxt ("src","fig:title")]
```

Represents an image with a caption. This is an special case of `Figure` but can only contain a single image.

Written as a figure this would be:

```
Figure attr (Caption Nothing [Plain plainTxt]) [Plain [Image attr plainTxt ("src", "title")]]
```

# `Table`

We haven't work on those in this project but they are relevant because tables *are* figures. Where in this case we are thinking of `Figure` as a floating element of the document.

**Comment by cderv** [comment]
Thanks that is useful. 

Having all this kind of information put in a single doc would be useful at some point !
